policy-controller/k8s/index/src/index.rs

```diff
@@ -1120,6 +1120,22 @@ impl PolicyIndex {
                 );
             }
         }
+        for tgt in spec.authentications.iter() {
```

Comment:
Can we combine these two loops and process both MeshTLS and ServiceAccount in the same loop?

```
for tgt in spec.authentications.iter() {
  match tgt {
    AuthenticationTarget::MeshTLS { ref namespace, ref name } => { ... },
    AuthenticationTarget::ServiceAccount { ref namespace, ref name } => { ... },
    AuthenticationTarget::Network { ref namespace, ref name } => { ... },
  }
}
```